ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/dhcp_servers.schema.yml

```
@@ -0,0 +1,98 @@
+# Copyright (c) 2023 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+# yaml-language-server: $schema=../../../../plugins/plugin_utils/schema/avd_meta_schema.json
+# Line above is used by RedHat's YAML Schema vscode extension
+# Use Ctrl + Space to get suggestions for every field. Autocomplete will pop up after typing 2 letters.
+type: dict
+keys:
+  dhcp_servers:
+    type: list
+    items:
+      type: dict
+      keys:
+        disabled:
+          type: bool
+        vrf:
+          type: str
+        dns_domain_name_ipv4:
+          type: str
+        dns_domain_name_ipv6:
+          type: str
+        ipv4_vendor_options:
+          type: list
+          items:
+            type: dict
+            keys:
+              vendor_id:
+                type: str
+              sub_options:
+                type: list
+                items:
+                  type: dict
+                  keys:
+                    number:
+                      type: int
+                      required: true
+                      min: 1
+                      max: 254
+                      convert_types:
+                      - str
+                    type:
+                      type: str
+                      required: true
+                      valid_values: ["string", "ipv4-address", "array ipv4-address"]
+                    data:
+                      type: str
+                      required: true
+        subnets:
+          type: list
+          items:
```

```suggestion
        subnets:
          type: list
          primary_key: subnet
          items:
```

(NOTE that it could be abused as we would consider `fe80::/64` and `fe80:0::/64` as different string , but they would be the same on the device .. :) but better than nothing)

---

Actually ... ipv4 and ipv6 subnets don't have the same options
so let's call this `ipv4_subnets`

IPv6 (no default-gateway)
```
ceos2(config-dhcp-subnet-ipv6)#show cli commands
APPEND [ COMMENT ]
[no|default] ( client class ) ( definition CLIENT_CLASS )
[no|default] client class assignment { CLIENT_CLASS }
[no|default] comment
[no|default] dns server { SERVERS }
do CMD
exit
[no|default] lease time ( DAYS days HOURS hours MINUTES minutes )
[no|default] name NAME
[no|default] range START END
[no|default] reservations
show active
show active all [ detail ]
show comment
[no|default] ( tftp server ) file FILENAME
ceos2(config-dhcp-subnet-ipv6)#exit
```

IPv4 (with default-gateway)
```
ceos2(config-dhcp-subnet-ipv4)#show cli commands
APPEND [ COMMENT ]
[no|default] ( client class ) ( definition CLIENT_CLASS )
[no|default] comment
[no|default] dns server { SERVERS }
do CMD
exit
[no|default] default-gateway IP_ADDR
[no|default] lease time ( DAYS days HOURS hours MINUTES minutes )
[no|default] name NAME
[no|default] range START END
[no|default] reservations
show active
show active all [ detail ]
show comment
[no|default] ( tftp server ) file FILENAME
[no|default] ( tftp server option 150 ) { SERVER_ADDR }
[no|default] ( tftp server option 66 ) NAME
```